src/traces/scattergl/convert.js

```diff
@@ -511,7 +593,11 @@ proto.updateFancy = function(options) {
             this.scatter.options.borderWidths[i] = 0.5 * borderWidths[index];
 
             for(j = 0; j < 4; ++j) {
-                this.scatter.options.colors[4 * i + j] = colors[4 * index + j];
+                var color = colors[4 * index + j];
+                if(sel && !sel[index] && j === 3) {
+                    color *= DESELECTDIM;
+                }
+                this.scatter.options.colors[4 * i + j] = color;
```

Comment:
Did you get the time to investigate :arrow_heading_up: ? If not, I can look at it this afternoon.